app/uk/gov/hmrc/uknwauthcheckerapi/generators/RandomGen.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.uknwauthcheckerapi.generators
+
+import org.scalacheck.Gen
+import wolfendale.scalacheck.regexp.RegexpGen
+
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+import scala.util.Random
+
+trait RandomGen {
```

Comment:
Good work on this, although can you move it out of the api package and into a common package so it can be used by both the IT tests and unit tests.

This example uses a tests-common package and it's shared between it & tests packages.

https://github.com/hmrc/economic-crime-levy-returns/tree/main/test-common/uk/gov/hmrc/economiccrimelevyreturns